### PR TITLE
feat!: Align `path_view::render_*` APIs with P1036R6

### LINKED ISSUE #139: `render` functions do not match P1030R6
In [P1030R5](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p1030r5.pdf) the render functions were named `render_zero_terminated`/`render_not_zero_terminated` and took an additional `path_view_component` as first argument.

In [P1030R6](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1030r6.pdf) the names were changed to `render_null_terminated`/`render_unterminated` and the unnecessary first argument was removed.

[This commit](https://github.com/ned14/llfio/commit/938b8456e9d575c4455ddeca5fbe808c040b8c88) changed the names to the P1030R6 convention, however it left the first argument there.

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
Resolves #139

### CODE REVIEW

**Comment by BurningEnlightenment** [comment]
@ned14 can you investigate the CI failures?
* Programs / Programs (windows-2019): Tries to compile quickcpplib with valgrind support which obviously doesn't work.
* Installability / Installability (macos-latest, *: Seems to hang during outcome installation.

File: include/llfio/v2.0/path_view.hpp [hunk 1]
```diff
@@ -1568,21 +1568,19 @@ class LLFIO_DECL path_view_component
   LLFIO_TREQUIRES(LLFIO_TPRED(is_source_acceptable<T>),
```
Comment (sneves) [inline include/llfio/v2.0/path_view.hpp:1568]: `render()`, immediately above this, was removed in P1036R6. Consistency dictates that either it should go away or also remove the first argument.

Comment (BurningEnlightenment) [reply include/llfio/v2.0/path_view.hpp:1568]: Indeed, I missed the history entries
* R5: `render()` function added as per LEWG request
* R6: `render()` function removed as per LEWG request

🙃

However, given that `render()` has existed for almost two years, I would first deprecate it for one "release" and then remove it. That should ease the transition for our user base considerably at a very low cost.

**Comment by ned14** [comment]
Programs should now be fixed on Windows. Sorry about that.

The macos installability tests take a very long time. I actually axed a bunch of stuff to make them go much faster. Even then, they are hideously slow. I assume it's an x64 box emulating a Mac M1 CPU.

I do note github actions prints a bunch of warnings about the actions being used being too old, and therefore it will only run on them node X or whatever. Which _could_ mean "emulated slow as mollasses" but I haven't investigated.

File: include/llfio/v2.0/path_view.hpp (hunk 1 above)
Comment (ned14) [reply include/llfio/v2.0/path_view.hpp:1568]: I think the number of users of those functions is likely very low, as they were only added by LEWG and as you correctly point out, were clearly added wrong by me likely very late at night when my eyes weren't working.

P1030R7 is nearing completion after a month of working on it, so I'm going to go ahead and merge this. Thanks for the changeset.

**Pull request merged**